diff --git a/CMakeLists.txt b/CMakeLists.txt
deleted file mode 100644
index ddff601..0000000
--- a/CMakeLists.txt
+++ /dev/null
@@ -1,26 +0,0 @@
-cmake_minimum_required(VERSION 3.8)
-
-set(PROJECT_NAME list-devices)
-project(${PROJECT_NAME} CXX)
-
-if (NOT CMAKE_BUILD_TYPE) 
-  set(CMAKE_BUILD_TYPE Release) 
-endif() 
-
-set(CMAKE_CXX_FLAGS "-Wall -Wextra -pedantic")
-set(CMAKE_CXX_FLAGS_DEBUG "-g")
-
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
-
-include_directories(include)
-
-link_directories(${CMAKE_PREFIX_PATH}/lib/)
-
-file(GLOB SOURCES "${PROJECT_SOURCE_DIR}/src/*.cpp")
-add_executable(${PROJECT_NAME} ${SOURCES})
-
-target_compile_options(${PROJECT_NAME} PRIVATE -O3)
-
-target_link_libraries(${PROJECT_NAME} v4l2)
diff --git a/include/list_devices.hpp b/include/v4l2_list_devices/list_devices.hpp
similarity index 98%
rename from include/list_devices.hpp
rename to include/v4l2_list_devices/list_devices.hpp
index db3414e..db731d2 100644
--- a/include/list_devices.hpp
+++ b/include/v4l2_list_devices/list_devices.hpp
@@ -20,6 +20,9 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <stdio.h>
+// #fixed - added these two missing includes
+#include <cstring>
+#include <stdexcept>
 
 #include <string>
 #include <vector>
