diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..a255e13
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.5.1)
+
+project(whereami C CXX)
+
+add_library(${PROJECT_NAME} STATIC "${CMAKE_CURRENT_SOURCE_DIR}/src/whereami.c")
+
+target_include_directories(${PROJECT_NAME}
+    PUBLIC
+        "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>"
+)

diff --git a/src/whereami.c b/src/whereami.c
index a8a649d..886eaad 100644
--- a/src/whereami.c
+++ b/src/whereami.c
@@ -118,10 +118,12 @@ static int WAI_PREFIX(getModulePath_)(HMODULE module, char* out, int capacity, i
     if (!_wfullpath(buffer2, path, MAX_PATH))
       break;
     length_ = (int)wcslen(buffer2);
-    length__ = WideCharToMultiByte(CP_UTF8, 0, buffer2, length_ , out, capacity, NULL, NULL);
+    // use CP_ACP to match LabVIEW's on Windows behavior
+    length__ = WideCharToMultiByte(CP_ACP, 0, buffer2, length_ , out, capacity, NULL, NULL);
 
     if (length__ == 0)
-      length__ = WideCharToMultiByte(CP_UTF8, 0, buffer2, length_, NULL, 0, NULL, NULL);
+    // use CP_ACP to match LabVIEW's on Windows behavior
+      length__ = WideCharToMultiByte(CP_ACP, 0, buffer2, length_, NULL, 0, NULL, NULL);
     if (length__ == 0)
       break;
 
