diff --git a/CMakeLists.txt b/CMakeLists.txt
index 97cd0ca..b4f4667 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,11 +56,6 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
         )
     endif()
     add_compile_options(-Wno-shift-negative-value)
-    if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24.0")
-      set(CMAKE_COMPILE_WARNING_AS_ERROR ON)
-    else()
-      add_compile_options(-Werror)
-    endif()
 endif()
 
 if(CMAKE_C_COMPILER_ID MATCHES "Clang" AND CMAKE_C_SIMULATE_ID MATCHES "MSVC")
